### PR TITLE
Rails Blog app - controllers specs

### PR DESCRIPTION
**Pull request by killy10o10**
# Description:
In this feature, I set up tests on the controllers I recently added to the Rails Blog app.

## Completed the following Project requirements
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
+      comments_counter: 0,
+      likes_counter: 0
+    )
+    @post.save
+
+    describe 'GET /index' do
+      it 'renders the index template' do
+        get '/users/:id/posts/'
+        expect(response.status).to eq(200)
+        expect(response.body).to include('List all users posts')
```
Comment (Ismailco) [inline spec/requests/posts_spec.rb]: - [x] Kindly make sure to include tests that ensure the correct view template is rendered, as you did in the other action test. e.g `expect(response).to render_template('index')`

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
+      get "/users/#{@user.id}"
+      expect(response.status).to eq(200)
+      expect(response).to render_template('show')
+      expect(response.body).to include('Users#show')
+    end
+  end
+end
 
 RSpec.describe 'Users', type: :request do
   describe 'GET /index' do
```
Comment (Ismailco) [inline spec/requests/users_spec.rb]: - [x] To Not repeat yourself and follow the best practices kindly make sure to combine your tests in one `Rspec. describe....`

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -1,4 +1,34 @@
...
+      posts_counter: 0
+    )
+    @user.save
+  end
+
+  describe 'GET /index' do
+    it 'renders index template' do
+      get '/users/'
+      expect(response.status).to eq(200)
+      expect(response).to render_template('index')
```
Comment (Ismailco) [inline spec/requests/users_spec.rb:19]: - [x] Great job creating the tests for users requests however you should include tests that ensure the response body includes the correct placeholder text, you did this in the next test but kindly make sure to include it also in this test. e.g `expect(response.body).to include(.....)`

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
 
-  describe 'GET /show' do
-    it 'returns http success' do
-      get '/posts/show'
-      expect(response).to have_http_status(:success)
+    describe 'GET /show' do
+      it 'renders the show template' do
+        get "/users/posts/#{@post.id}"
+        expect(response.status).to eq(200)
+        expect(response).to render_template('show')
```
Comment (Ismailco) [inline spec/requests/posts_spec.rb:27]: - [x] Great job creating the tests for `posts` requests however you should include tests that ensure the response body includes the correct placeholder text, you did this in the previous test but kindly make sure to include it also in this test. e.g `expect(response.body).to include(.....)`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @killy10o10, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @killy10o10 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**